src/actions/post_build.rs

```diff
@@ -86,8 +86,11 @@ impl PostBuildHandler {
                 trace!("build - Error {} when running {:?}", cause, cmd);
                 self.notifier.notify_begin_diagnostics();
                 if !self.shown_cargo_error.swap(true, Ordering::SeqCst) {
-                    let msg = format!("There was an error trying to build, RLS features will be limited: {}", cause);
-                    self.notifier.notify_error_diagnostics(&msg);
+                    // It's not a good idea to make a long message here, the output in
+                    // VSCode is one single line, and it's important to capture the
+                    // root cause.
+                    // let msg = format!("There was an error trying to build, RLS features will be limited: {}", cause);
+                    self.notifier.notify_build_error_diagnostics(cause);
```

Comment:
It sounds like a good idea to make the message shorter, but we probably need some boilerplate here - the `cause` message might not be very informational if it is an internal error message and we should make clear to the user that something failed.

Reply:
Yes. It's a hard balance. I think `Cargo failed:` that comes from the underlying error addresses that somewhat. I don't know if there is some other mechanism we can use to provide more details to the user?